app/src/main/java/com/example/weather/WeatherService.java

```diff
@@ -14,66 +14,67 @@
 
 public class WeatherService extends Service {
     private static final String TAG = WeatherService.class.getSimpleName();
+    private static final String processName = Application.getProcessName();
 
     private ExecutorService executorService;
 
-    private final IBinder binder = new WeatherBinder();
-
-    private Handler handler;
+    IWeatherService.Stub binder = new IWeatherService.Stub() {
+        @Override
+        public void getCurrentWeatherInfo(WeatherRequest weatherRequest, IWeatherServiceCallback callback) {
+            Log.i(TAG, "make binder");
+            WeatherProviderRunnable weatherProviderRunnable = new WeatherProviderRunnable(weatherRequest, callback);
+            executorService.execute(weatherProviderRunnable);
```

Comment:
Move out this code to the Service level method.
```suggestion
            WeatherService.this.getCurrentWeatherInfo(request, callback);
```